teps/0050-ignore-task-failures.md

```diff
@@ -187,6 +194,220 @@ control over the task definitions but may desire to ignore a failure and continu
 
   ![Jenkins Dashboard](images/0050-jenkins-dashboard-with-failure-stage.png)
 
+## Proposal
+We propose a new filed ```OnError``` to the [PipelineTask](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#adding-tasks-to-the-pipeline) definition.
+
+```go
+type PipelineTask struct {
+	Name string `json:"name,omitempty"`
+
+	//...
+
+	// OnError defines the termination behavior of a pipeline when the task is on error
+	// can be set to [ continue | stopAndFail ]
+	OnError OnErrorType `json:onError, "omitempty"`
+}
+```
+
+```go
+type OnErrorType string
+
+const (
+	// StopAndFail indicates to stop the pipeline if the task is failed
+	StopAndFail OnErrorType = "stopAndFail"
+	// Continue indicates to continue executing the rest of the pipeline irrespective of the status of the task
+	Continue    OnErrorType = "continue"
+)
+```
+
+Pipeline author can set the ```OnError``` field to configure the task failure strategy. If set to ```StopAndFail```, the pipeline is stopped and failed when the task is failed. If set to ```Continue```, the failure of task is ignored and the pipeline continues to execute the rest of the DAG.
+
+```yaml
+- name: task1
+  onError: continue
+  taskSpec:
+    steps:
+      - image: alpine
+        name: exit-with-1
+        script: |
+          exit 1
+
+```
+This new field ```OnError``` will be implemented as a alpha feature and can be enabled by setting enable-api-fields to alpha.
+
+Setting ```OnError``` is optional, the default pipeline behaviour is ```StopAndFail```
+
+The task run information is avaialble under the ```pipelineRun.status.taskRuns```. Note that the original task run status remains as it is irrelevant of the value of ```OnError``` (i.e. a failed task with ```onError: continue``` is still marked as failed). The task would be considered "successful" ONLY for the purposes of determining the status of the pipeline, which is represented in ```pipelineRun.status.conditions``` 
```

Comment:
note that we're moving to `pipelineRun.status.childReferences` as was proposed in TEP-0100:
https://github.com/tektoncd/community/blob/main/teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

Reply:
Thanks for pointing out! I added explanation for both scenarios